src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -694,6 +696,11 @@ private static void Write(string val, byte[] bytes, FileStream stream)
         private void Serialize(string filename)
         {
             AnalysisCacheData fromOtherProcess = null;
+            if (!_saveCacheToDisk)
```

Comment:
I think this check can also be removed, though it wouldn't hurt to assert here.

The design is that nobody can call `Serialize` without going through `QueueSerialization`, so the check isn't needed here.